### ISSUE TITLE
packageRules.matchCurrentAge

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#21368]
### What would you like Renovate to be able to do?

Support a new feature `matchCurrentAge` in `packageRules`.

### If you have any ideas on how this should be implemented, please tell us here.

The age value should be specified using the "pretty time" utility already written: https://github.com/renovatebot/renovate/blob/main/lib/util/pretty-time.ts

A valid value for this field should start with `<`, `<=`, `>` or `>=` and then be followed by a valid duration string, e.g. `> 3 years`.

The "current age" itself should be based on the release timestamps from the datasource. In other words such a package rule can never match until the datasource lookup has completed, but it should then be matched and applied prior to options like `separateMajorMinor` being evaluated.

### Is this a feature you are interested in implementing yourself?

No

**Comment by RahulGautamSingh** [comment]
It can be applied after we compute the `currentVersion` and use it to find the `releaseTimestamp`. WDYT?
https://github.com/renovatebot/renovate/blob/c89eeb89dff3ee5a62b71c936bd447e579512e27/lib/workers/repository/process/lookup/index.ts#L207

**Comment by rarkins** [comment]
Yes, first you need the `currentVersion`, then you can derive a currentReleaseTimestamp

**Comment by RahulGautamSingh** [comment]
Exact dates allowed aren't allowed in `matchCurrentAge`, right? `1 year | 1 day`.

**Comment by rarkins** [comment]
I'm not sure I understand your question, but I had expected you could say the equivalent of "older than 1 year and 1 day"

**Comment by RahulGautamSingh** [comment]
My bad, the eg is confusing 😅. I meant to ask if this will be okay?
```json
{
    matchCurrentAge : "3 years"
}
```
A duration string without any of the following operators `>|<|>=|<=`.

**Comment by RahulGautamSingh** [comment]
@rarkins , we're looking to categorize packages based on the age of their current version, right? 

Take `chalk` for example: 
Say, its current version is `4.1.1`, which is over 3 years old. Other available versions are `4.1.2` which is 2 years old,  and `5.3.0` which is over 5 months old.

```json
{
  "packageRules": [
    {
      "matchCurrentAge": "> 1 years",
      "labels": ["old-ones"]
    }
  ]
}
```

With this configuration:

Are we aiming to add the label `old-ones` to all update PRs for the chalk package since its current version `4.1.1` is more than 3 years old and it satisfies the `> 1 years ` range? 

Or, 

Are we looking at applying the label only to those update PRs that are updating the package to a new version which is more than 1 year old? 
For instance, this would mean labeling updates like chalk to `4.1.2`, but not for updates to `5.3.0`.

**Comment by rarkins** [comment]
@RahulGautamSingh it's intended to for the first case, i.e. about the age of the current version, and not about the age or age difference of updates
